src/test/java/com/blog/stream/pagination/PageSpliteratorNonJpaTest.java

```diff
@@ -0,0 +1,235 @@
+package com.blog.stream.pagination;
+
+import org.assertj.core.util.Sets;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PageSpliteratorNonJpaTest {
+
+    static Logger LOG = LoggerFactory.getLogger(PageSpliteratorNonJpaTest.class);
+
+    PagedRepository<TestObject> testObjectPagedRepository;
+    List<TestObject> source;
+
+    TestObject createTestObject(int id) {
+        return new TestObject(id, String.valueOf(id));
+    }
+
+    @Before
+    public void setup() {
+        source = IntStream.range(0, 1000).mapToObj(this::createTestObject).collect(Collectors.toList());
+        testObjectPagedRepository = new PagedRepository<>(source);
+    }
+
+    @Test
+    public void trySplitShouldReturnChildWithCurrentPageAndMoveOnToNextPage() {
+        PageSpliterator<PagedResult<TestObject>, TestObject> spliterator = PageSpliterator.create(3, 100, 10, null, null);
+        Spliterator<TestObject> child = spliterator.trySplit();
+
+        Assert.assertEquals(3, ((PageSpliterator.ChildPageSpliterator<PagedResult<TestObject>, TestObject>) child).getPageNumber());
+        Assert.assertEquals(4, spliterator.getPageNumber());
+
+    }
+
+    @Test
+    public void trySplitAndForEachRemainingShouldIterateAndReturnAllObjects() {
```

Comment:
I should probably add a Contributers file, never expected anyone to want to use this :D 

The format ive been using for test methods is `methodUnderTest_Scenario_ExpectedOutcome`

Reply:
Maybe `trySplit_TotalCount1000pageSize100_creates10Spliterators`?